pex/cli/commands/lock.py

```diff
@@ -119,6 +131,237 @@ def __call__(self, parser, namespace, value, option_str=None):
         setattr(namespace, self.dest, dry_run_style)
 
 
+@attr.s(frozen=True)
+class SyncTarget(object):
+    @classmethod
+    def resolve_command(
+        cls,
+        venv,  # type: Virtualenv
+        command=None,  # type: Optional[Tuple[str, ...]]
+    ):
+        # type: (...) -> SyncTarget
+        if command:
+            argv0 = command[0] if os.path.isabs(command[0]) else venv.bin_path(command[0])
+            command = (argv0,) + command[1:]
+        return SyncTarget(venv=venv, command=command)
+
+    @classmethod
+    def resolve_venv(
+        cls,
+        argv0,  # type str
+        *additional_args  # type: str
+    ):
+        # type: (...) -> Union[SyncTarget, Error]
+
+        argv0_path = None  # type: Optional[str]
+        if is_exe(argv0):
+            argv0_path = argv0
+        else:
+            path = os.environ.get("PATH")
+            if path:
+                for entry in path.split(os.pathsep):
+                    exe_path = os.path.abspath(os.path.join(entry, argv0))
+                    if is_exe(exe_path):
+                        argv0_path = exe_path
+                        break
+
+        if argv0_path:
+            venv = Virtualenv.enclosing(python=argv0_path)
+            if not venv:
+                try:
+                    venv = Virtualenv(os.path.dirname(os.path.dirname(argv0_path)))
```

Comment:
I see that this double parent is used a few places in the code base so probably isn't "new"... what it's unclear what it means? What `--venv` argument value might a user pass that falls into this path?

Reply:
The venv bin dir. So /a/venv/bin/script. The parent is /a/venv/bin/, and its parent is /a/venv/ - the venv dir.

Reply:
N.B.: For Windows the structure is `C:\a\venv\Scripts\script.exe` and so the `..\..` treatment holds.